keycloak/util.go

```diff
@@ -1,11 +1,70 @@
 package keycloak
 
 import (
+	"bytes"
 	"strconv"
 	"strings"
 	"time"
 )
 
+type KeycloakBoolQuoted bool
+type KeycloakBool bool
```

Comment:
I think I understand what these are for, but let me just confirm:

- `KeycloakBool` can be either `""` (false), `true`, or `false`.
- `KeycloakBoolQuoted` can be either `""` (false), `"true"`, or `"false"`. This is used in "config" API objects where all values are in quotes.

Is this correct?

If so, could you help me understand why `KeycloakBool` is needed?  It looks like you used it in a few places in the generic identity provider helpers, but it looks like those APIs always return booleans.  I haven't seen any of them return empty quotes (running v4.8.3.Final).

Reply:
I tested with keycloak 4.5.0 and earlier and all boolean values are just empty quotes if they are default and quoted boolean values if they aren't

Reply:
I took a closer look and it looks like attributes using `KeycloakBool` are only empty strings when using the console to create them with default values.  All subsequent fetches seem to be actual booleans instead of strings.

So I don't _think_ `KeycloakBool` is needed, but I am fine with leaving it if you think it is necessary.

This wouldn't really be an issue if Keycloak had a more consistent REST API :smile: